src/libos/src/config.rs

```diff
@@ -465,4 +516,146 @@ struct InputConfigMountOptions {
     pub temporary: bool,
     #[serde(default)]
     pub total_size: Option<String>,
+    #[serde(default)]
+    pub index: u32,
+    #[serde(default)]
+    pub autokey_policy: Option<u32>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+struct InputConfigApp {
+    #[serde(default)]
+    pub stage: String,
+    #[serde(default)]
+    pub entry_points: Vec<String>,
+    #[serde(default)]
+    pub mount: Vec<InputConfigMount>,
+    #[serde(default)]
+    pub encrypted: bool,
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+#[allow(non_camel_case_types)]
+pub struct user_rootfs_config {
+    // length of the struct
+    len: usize,
+    // UnionFS type rootfs upper layer, read-write layer
+    upper_layer_path: *const i8,
+    // UnionFS type rootfs lower layer, read-only layer
+    lower_layer_path: *const i8,
+    entry_point: *const i8,
+    // HostFS source path
+    hostfs_source: *const i8,
+    // HostFS target path, default value is "/host"
+    hostfs_target: *const i8,
+    // An array of pointers to null-terminated strings
+    // and must be terminated by a null pointer
+    envp: *const *const i8,
+}
+
+fn to_option_pathbuf(path: *const i8) -> Result<Option<PathBuf>> {
+    let path = if path.is_null() {
+        None
+    } else {
+        Some(PathBuf::from(
+            from_user::clone_cstring_safely(path)?
+                .to_string_lossy()
+                .into_owned(),
+        ))
+    };
+
+    Ok(path)
+}
+
+impl ConfigApp {
+    pub fn from_user(config: &user_rootfs_config) -> Result<ConfigApp> {
+        // Check config struct length for future possible extension
+        if config.len != size_of::<user_rootfs_config>() {
+            return_errno!(EINVAL, "User Config Struct length not match");
+        }
+
+        // Update INIT_ENVS if necessary
+        if !config.envp.is_null() {
```

Comment:
I think it is possible to merge `init env` and `default env` to ConfigEnv, and rename ConfigEnv.default to trusted, what do you think?

Reply:
Rename "default" to “trusted" makes sense. But this "default" corresponds to the name in Occlum.json. To make two consistent, it would be better to rename them together. While this change may require many existed demos change accordingly as well which is not so friendly to the Occlum users. So I think we can keep the "default" name as it is.